examples/gno.land/r/demo/gnofundme/gnofundme.gno

```diff
@@ -0,0 +1,319 @@
+package gnofundme
+
+import (
+	"bytes"
+	"std"
+	"strconv"
+	"strings"
+	"time"
+
+	"gno.land/p/demo/avl"
+	GFMpkg "gno.land/p/demo/gnofundme" // GnoFundMe package
+	"gno.land/p/demo/ufmt"
+)
+
+var (
+	campaigns      *avl.Tree
+	users          *avl.Tree
+	banker         std.Banker
+	balance        int64
+	campaignPrefix = "/r/demo/gnofundme:"
+	campaignsTotal = make(map[string]int64)
+)
+
+func init() {
+	campaigns = avl.NewTree()
+	users = avl.NewTree()
+	banker = std.GetBanker(std.BankerTypeRealmIssue)
+	// banker.IssueCoin(std.CurrentRealm().Addr(), "ugnot", 350000000)
+}
+
+func NewUser(name string, email string) string {
+	// get the user address
+	address := std.GetOrigCaller()
+	// create the user
+	user := GFMpkg.NewUser(name, email, address)
+
+	// the avl.tree key will be the address to simplify the fetching
+	users.Set(address.String(), user)
+	std.Emit("User Creation", "user_created", user.GetAddress().String())
+	return user.GetName() + " added successfully"
+}
+
+func NewCampaign(ctype int, title string, description string, goal int64, beginT int64, deadlineT int64, beneficiary std.Address) string {
+	if ctype < 0 || ctype > 5 {
+		panic("invalid campaign type")
+	}
+
+	var owner, beneficiaryU *GFMpkg.User
+	// gets the caller address (owner of the campaign)
+	caller := std.GetOrigCaller()
+
+	// verify that the user owner exists
+	o, ok := users.Get(caller.String())
+	if !ok {
+		// panic("owner not found")
+		// XXX TO BE MODIFIED: temporarily creates the user if he doesn't exist. In the final version,
+		// the user will have to be registered using the user struct of this project, or maybe I'll use
+		// the profile pkg that is still a "work in progress"
+		user := GFMpkg.NewUser("user1", "email1", caller)
+		// the avl.tree key will be address to simplify the fetching
+		users.Set(caller.String(), user)
+		owner = user
+	} else {
+		// cast the owner and beneficiary to *GFMpkg.User
+		owner = o.(*GFMpkg.User)
+	}
+
+	// verify that the benefeciary exists if it's different from caller
+	if beneficiary != caller {
+		b, ok := users.Get(beneficiary.String())
+		if !ok {
+			// panic("beneficiary not found")
+			user := GFMpkg.NewUser("user2", "email2", beneficiary)
+			users.Set(beneficiary.String(), user)
+			beneficiaryU = user
+		} else {
+			beneficiaryU = b.(*GFMpkg.User)
+		}
+	} else {
+		beneficiaryU = owner
+	}
+
+	// Verfiy that the dates are correct
+	if beginT < time.Now().Unix() {
+		panic("begin date must be in the future")
+	}
+	if deadlineT < time.Now().Unix() {
+		panic("deadline date must be in the future")
+	}
+	if deadlineT < beginT {
+		panic("deadline date must be after begin date")
+	}
+
```

Comment:
This is redundant and can be simplified; you can check these two things:
- if `beginT` is in the future
- if `deadlineT` is after `beginT`

With transitivity, if:
`now < begin && begin < deadline
=> now < deadline
`
Also, `begin` could be `>= now`